File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.js
function solution(food) {
    //let foodArr = food.slice(1,food.length);
    // 한 선수가 먹는 음식의 배열
    let first = [];
    // 다른 선수가 먹는 음식의 배열(한 선수의 배열을 복사해서 내림차순으로 정렬)
    let second = [];
    for(let i=1; i<food.length; i++){
        let count = Math.floor(food[i]/2);
        for(let j=0; j<count; j++){
            first.push(i);
            second.push(i);
        }
    }
    second.sort((a,b) => b-a);
    first.push(0);
    let result = first.concat(second).join("");
    return result;
    
    
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
def solution(food):
    left_side = ''
    for i in range(1, len(food)):
        count = food[i] // 2
        left_side += str(i) * count
    
    right_side = left_side[::-1]

    answer = left_side + '0' + right_side
    
    return answer
    
# 2로 나눴을때 몫 만큼 배치하고 0 , 나머지는 뒤집어서 붙임
